src/v/cloud_storage/offset_translation_layer.h

```diff
@@ -50,12 +55,9 @@ class offset_translator final {
       ss::output_stream<char> dst,
       retry_chain_node& fib) const;
 
-    /// Get segment name adjusted for all removed offsets
-    segment_name get_adjusted_segment_name(
-      const segment_meta& s, retry_chain_node& fib) const;
-
 private:
     model::offset_delta _initial_delta;
+    ss::lw_shared_ptr<storage::offset_translator_state> _ot_state;
```

Comment:
nit: could we get the offset translator to own the state (i.e. use a `unique_ptr` instead)?

Reply:
it shouldn't be owned because it's passed down the line after all segments are downloaded